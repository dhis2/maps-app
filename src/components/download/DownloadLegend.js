import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Legend from '../layers/legend/Legend';
import { legendPositions } from './LegendPosition';

const styles = theme => ({
    root: {
        position: 'absolute',
        paddingTop: theme.spacing.unit,
        paddingLeft: theme.spacing.unit,
        paddingRight: theme.spacing.unit,
        zIndex: 998,
        backgroundColor: 'rgba(255,255,255,0.9)',
        boxShadow: theme.shadows[1],
        borderRadius: theme.shape.borderRadius,
        fontSize: theme.typography.fontSize,
        maxWidth: 220,
        '& div a:link, & div a:visited': {
            color: theme.palette.text.primary,
        },
    },
    legend: {
        marginBottom: 16,
    },
    title: {
        fontSize: 15,
        fontWeight: 500,
        margin: 0,
        paddingBottom: 10,
    },
    period: {
        display: 'block',
        fontWeight: 'normal',
    },
    topleft: {
        top: theme.spacing.unit,
        left: theme.spacing.unit,
    },
    topright: {
        top: theme.spacing.unit,
        right: theme.spacing.unit,
    },
    bottomleft: {
        bottom: 30,
        left: theme.spacing.unit,
    },
    bottomright: {
        bottom: 20,
        right: theme.spacing.unit,
    },
    name: {
        top: theme.spacing.unit * 6,
    },
});

export const DownloadLegend = ({ position, layers, showName, classes }) => {
    let className = `${classes.root} ${classes[position]}`;

    if (showName && position.substring(0, 3) === 'top') {
        className += ` ${classes.name}`;
    }

    const legends = layers
        .filter(layer => layer.legend)
        .map(layer => layer.legend)
        .reverse();

    return (
        <div className={className}>
            {legends.map((legend, index) => (
                <div key={index} className={classes.legend}>
                    <h2 className={classes.title}>
                        {legend.title}
                        <span className={classes.period}>{legend.period}</span>
                    </h2>
                    <Legend {...legend} />
                </div>
            ))}
        </div>
    );
};

DownloadLegend.propTypes = {
    position: PropTypes.oneOf(legendPositions).isRequired,
    layers: PropTypes.array.isRequired,
    showName: PropTypes.bool.isRequired,
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(DownloadLegend);
