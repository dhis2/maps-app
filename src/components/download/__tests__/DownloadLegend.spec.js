import React from 'react';
import { shallow } from 'enzyme';
import { DownloadLegend } from '../DownloadLegend';

describe('DownloadLegend', () => {
    const layers = [
        {
            id: 'layer-1',
            legend: {
                title: 'Layer 1',
                period: '2018',
            },
        },
        {
            id: 'layer-2',
        },
        {
            id: 'layer-3',
            legend: {
                title: 'Layer 2',
                period: '2017',
            },
        },
    ];

    const renderComponent = props =>
        shallow(
            <DownloadLegend
                position="bottomright"
                layers={[]}
                showName={false}
                {...props}
            />
        );

    it('Should render a legend component', () => {
        expect(renderComponent().exists()).toBe(true);
    });

    it('Should only render layers with legends', () => {
        const wrapper = renderComponent({ layers });
        expect(wrapper.children().length).toEqual(2);
        expect(wrapper.find('Legend').length).toEqual(2);
    });

    it('Should render legend title and period', () => {
        const wrapper = renderComponent({ layers });
        expect(wrapper.containsMatchingElement('Layer 1')).toBe(true);
        expect(wrapper.containsMatchingElement(<span>2018</span>)).toBe(true);
    });
});
