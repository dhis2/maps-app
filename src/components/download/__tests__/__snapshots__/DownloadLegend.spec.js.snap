// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadLegend Should render a legend component 1`] = `
<div>
  <div
    class="legend"
  >
    <h2
      class="title"
      data-test="download-legend-title"
    >
      Layer 2
      <span
        class="period"
      >
        2017
      </span>
    </h2>
    <dl
      class="legend"
      data-test="layerlegend"
    />
  </div>
  <div
    class="legend"
  >
    <h2
      class="title"
      data-test="download-legend-title"
    >
      Layer 1
      <span
        class="period"
      >
        2018
      </span>
    </h2>
    <dl
      class="legend"
      data-test="layerlegend"
    />
  </div>
</div>
`;
